moveit_ros/planning_interface/moveit_cpp/src/moveit_cpp.cpp

```diff
@@ -196,16 +196,12 @@ const ros::NodeHandle& MoveItCpp::getNodeHandle() const
 
 bool MoveItCpp::getCurrentState(moveit::core::RobotStatePtr& current_state, double wait_seconds)
 {
-  if (wait_seconds > 0.0 &&
-      !planning_scene_monitor_->getStateMonitor()->waitForCurrentState(ros::Time::now(), wait_seconds))
+  auto csm = planning_scene_monitor_->getStateMonitor();
+  if (!csm->waitForCurrentState(ros::Time::now(), wait_seconds) || !(current_state = csm->getCurrentState()))
```

Comment:
In general, I find it useful to separate the different potential sources of error rather than stacking them like this. I find that providing error messages that are specific to the actual failure makes for faster debugging which I find to be worth the cost of a few extra lines of code.

Reply:
Looking into the code, it seems that this doesn't actually change the default behavior so feel free to disregard my previous comment.